packages/medusa/src/services/draft-order.ts

```diff
@@ -297,62 +302,110 @@ class DraftOrderService extends TransactionBaseService {
         const lineItemServiceTx =
           this.lineItemService_.withTransaction(transactionManager)
 
-        for (const item of items || []) {
+        const itemsWithVariant: DraftOrderCreateProps["items"] = []
+        const itemsWithoutVariant: DraftOrderCreateProps["items"] = []
+
+        ;(items ?? []).forEach((item) => {
           if (item.variant_id) {
-            const line = await lineItemServiceTx.generate(
-              item.variant_id,
-              data.region_id,
-              item.quantity,
-              {
-                metadata: item?.metadata || {},
-                unit_price: item.unit_price,
-              }
-            )
+            itemsWithVariant.push(item)
+            return
+          }
 
-            await lineItemServiceTx.create({
-              ...line,
-              cart_id: createdCart.id,
-            })
-          } else {
+          itemsWithoutVariant.push(item)
+        })
+
+        const promises: Promise<any>[] = []
+
+        // generate line item link to a variant
+        if (itemsWithVariant.length) {
+          const itemsData = itemsWithVariant.map((item) => {
+            return {
+              variantId: item.variant_id,
+              quantity: item.quantity,
+              metadata: item.metadata,
+              unit_price: item.unit_price,
+            }
+          })
```

Comment:
thought(non-blocking): Can't we push this object structure directly into the array on line 310 instead of mapping them again?